### ISSUE TITLE
Support with in the presence of IApply patterns

### ISSUE THREAD

**Issue by andreasabel** [agda/agda#2461]
```agda
-- Requires Agda 2.6.0

{-# OPTIONS --cubical #-}

open import Agda.Primitive public

postulate
  _≡_   : ∀ {a} {A : Set a} → A → A → Set a
  PathP : ∀ {a} → (A : I → Set a) → A i0 → A i1 → Set a

{-# BUILTIN PATH  _≡_   #-}
{-# BUILTIN PATHP PathP #-}

-- With-clause stripping does not work with cubical yet:

refl : ∀{a}{A : Set a} (x : A) → x ≡ x
refl x i with x
... | q = x

-- An internal error has occurred. Please report this as a bug.
-- Location of the error: src/full/Agda/TypeChecking/Telescope.hs:377
```

**Comment by jespercockx** [comment]
We should do something about this before the release, at least there should be a proper error message that `with` is not supported for `--cubical`.

**Comment by Saizan** [comment]
We introduced an actual error message with the path constructors commit 188d57778aa8634b4536be9dfe5afe6658b9b341

```
with clauses not supported in the presence of Path patterns
when checking that the clause
refl x i with x
refl x i | q = x
has type {a : Level} {A : Set a} (x : A) → x ≡ x
```

This can be an enhancement now.

**Comment by Saizan** [comment]
Fixed by a6b452765e3e856865094b35b2b51f6da7c75a2a
